app/controllers/api/v1/users_controller.rb

```diff
@@ -3,7 +3,45 @@
 module API
   module V1
     class UsersController < API::V1::APIController
-      before_action :auth_user
+      before_action :auth_user, except: [:impersonate]
+      skip_before_action :authenticate_user!, only: [:impersonate]
+      skip_after_action :verify_authorized, only: :impersonate
+
+      def impersonate
+        # TODO use encryptorService para desencriptar queryParams
+        len = ActiveSupport::MessageEncryptor.key_len
+        salt, encrypted_data = params['queryParams'].split '$$'
+        encrypted_data = Base64.urlsafe_decode64(encrypted_data)
+        key = ActiveSupport::KeyGenerator.new(Rails.application.secrets.secret_key_base).generate_key(salt, len)
```

Comment:
>Initialize a new [MessageEncryptor](https://api.rubyonrails.org/classes/ActiveSupport/MessageEncryptor.html). secret must be at least as long as the cipher key size. For the default ‘aes-256-gcm’ cipher, this is 256 bits. If you are using a user-entered secret, you can generate a suitable key by using [ActiveSupport::KeyGenerator](https://api.rubyonrails.org/classes/ActiveSupport/KeyGenerator.html) or a similar key derivation function.
>
>The first additional parameter is used as the signature key for [MessageVerifier](https://api.rubyonrails.org/classes/ActiveSupport/MessageVerifier.html). This allows you to specify keys to encrypt and sign data. Ignored when using an AEAD cipher like ‘aes-256-gcm’.
>
>ActiveSupport::MessageEncryptor.new('secret', 'signature_secret')

https://api.rubyonrails.org/classes/ActiveSupport/MessageEncryptor.html#method-c-new


given that the message is built by us (no user entered) we can use just `ActiveSupport::MessageEncryptor.new(Rails.application.secret_key_base)` and avoid all the salt stuff?